src/h3lib/lib/algos.c

```diff
@@ -640,19 +662,23 @@ int H3_EXPORT(gridRingUnsafe)(H3Index origin, int k, H3Index *out) {
     // Scratch structure for checking for pentagons
     if (H3_EXPORT(isPentagon)(origin)) {
         // Pentagon was encountered; bail out as user doesn't want this.
-        return HEX_RANGE_PENTAGON;
+        return E_PENTAGON;
     }
 
     for (int ring = 0; ring < k; ring++) {
-        origin = h3NeighborRotations(origin, NEXT_RING_DIRECTION, &rotations);
+        H3Error neighborResult = h3NeighborRotations(
+            origin, NEXT_RING_DIRECTION, &rotations, &origin);
+        if (neighborResult) {
+            return neighborResult;
+        }
         if (origin == 0) {  // LCOV_EXCL_BR_LINE
```

Comment:
Nit: `H3_NULL`? Also, would `h3NeighborRotations` ever set `out` to `H3_NULL` now, or is this check meaningless?